examples/win-guid.c

```diff
@@ -83,9 +97,8 @@ status_t is_WINDOWS_KERNEL(vmi_instance_t vmi, addr_t base_p, uint8_t *pe)
     peparse_assign_headers(pe, NULL, NULL, &optional_header_type, &optional_pe_header, NULL, NULL);
     addr_t export_header_offset = peparse_get_idd_rva(IMAGE_DIRECTORY_ENTRY_EXPORT, &optional_header_type, optional_pe_header, NULL, NULL);
 
-    // The kernel's export table is continuously allocated on the PA level with the PE header
-    // This trick may not work for other PE headers (though may work for some drivers)
-    if ( base_p + export_header_offset < base_p + VMI_PS_4KB ) {
+    // Check if kernel's export table is within the PE image boundary
+    if ( base_p + export_header_offset < base_p + get_image_size(optional_header_type, optional_pe_header) ) {
```

Comment:
I would also add the export header size here.

Reply:
I've added a bound check for `et.name`:
```
et.name >= export_header_offset && et.name <= export_header_offset + export_header_size
```